test/h5test.c

```diff
@@ -2564,6 +2564,31 @@ h5_driver_uses_multiple_files(const char *drv_name, unsigned flags)
     return ret_val;
 }
 
+/*-------------------------------------------------------------------------
+ * Function:    h5_on_cygwin
+
+ *
+ * Purpose:     Determine whether tests are running on Cygwin
+ *
+ * Return:      True if running on Cygwin, false otherwise.
+ *
+ *-------------------------------------------------------------------------
+ */
+bool
+h5_on_cygwin()
+{
+    bool  ret_value = false;
+    FILE *fp        = NULL;
+
+    /* If /cygdrive/ exists and is readable, assume HDF5 is running on Cygwin */
+    if ((fp = fopen("/cygdrive", "r")) != NULL) {
```

Comment:
A perverse user could define this drive on a non-Cygwin system, but I'm not aware of any other way to check that Cygwin is in use besides the `__CYGWIN__` directive.